e2e/fixture/socket.ts

```diff
@@ -0,0 +1,55 @@
+import { type Page } from '@playwright/test';
+import { type Address } from 'viem';
+
+import { chain } from './base';
+
+function socket(page: Page) {
+  const assets = {
+    ETH: {
+      chainId: chain.id,
+      address: '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee',
+      name: 'Ethereum',
+      symbol: 'ETH',
+      decimals: 18,
+      chainAgnosticId: null,
+      icon: 'https://maticnetwork.github.io/polygon-token-assets/assets/eth.svg',
+      logoURI: 'https://maticnetwork.github.io/polygon-token-assets/assets/eth.svg',
+    },
+    OP: {
+      chainId: chain.id,
+      address: '0x4200000000000000000000000000000000000042',
+      name: 'Optimism',
+      symbol: 'OP',
+      decimals: 18,
+      chainAgnosticId: null,
+      icon: 'https://optimistic.etherscan.io/token/images/optimism_32.png',
+      logoURI: 'https://optimistic.etherscan.io/token/images/optimism_32.png',
+    },
+  };
+
+  type Balances = {
+    account: Address;
+    balances: {
+      symbol: 'ETH' | 'OP';
+      amount: number;
+    }[];
+  };
+
+  const balances = async (params: Balances) => {
+    await page.route(/api\.socket\.tech\/v2\/balances/, async (route) => {
```

Comment:
minor: why do we use a regex here? couldn't we just use a template string?

Reply:
Couldn't make it work, think it has to be the exact string which is filled with query parameters